File: src/main/java/de/sean/splugin/spigot/events/MoveEvent.java
package de.sean.splugin.spigot.events;

import de.sean.splugin.util.SMessages;
import de.sean.splugin.util.SUtil;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

public class MoveEvent implements Listener {
    @EventHandler
    public void PlayerMove(PlayerMoveEvent event) {
        final Player player = event.getPlayer();
        SUtil.setLastActivityForPlayer(player.getUniqueId(), System.currentTimeMillis());
        if (SUtil.isPlayerAFK(player.getUniqueId())) {
            SUtil.setPlayerAFK(player.getUniqueId(), false);
            SMessages.unmarkPlayerAFK(player);
        }
    }
}

File: spigot/src/main/kotlin/de/sean/splugin/bukkit/events/MoveEvent.kt
package de.sean.splugin.bukkit.events

import de.sean.splugin.util.SMessages.unmarkPlayerAFK
import de.sean.splugin.util.SUtil
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class MoveEvent : Listener {
    @EventHandler
    fun playerMove(event: PlayerMoveEvent) {
        val player = event.player
        SUtil.playerLastActivity[player.uniqueId] = System.currentTimeMillis()
        if (SUtil.afkPlayers[player.uniqueId]!!) {
            SUtil.afkPlayers[player.uniqueId] = false
            unmarkPlayerAFK(player)
        }
    }
}
